src/communication.ts

```diff
@@ -98,8 +98,12 @@ export class Communication {
     }
 
     // function that adds a new subscriber, which is called every time a message is received
-    subscribeMessage(subscriber: Subscriber): void {
-        this.socketServer.on('message', subscriber);
+    subscribeMessage(subscriber: Subscriber, type?: string): void {
+        this.socketServer.on('message', (message: Message) => {
+            if (message.getType() === type) {
+                subscriber(message);
+            }
+        });
```

Comment:
the socket server gives a string only you have to change the string to a message object
if there is no type provided subscribe to all messages 
all places the uses subscribe must be changed to expect a message instead of string since you changed iti